lib/querytables.js

```diff
@@ -175,14 +175,67 @@ function getForeignTableUpdatedAt (pg, table, callback) {
     }, readOnly);
 }
 
-/* For foreign tables we need to do extra queries to extract the updated_at properly */
-async function setUpdateAtToForeingTables (pg, rows) {
+function execQueryPromise ({ pg, sql, params, readOnly = true } = {}) {
+    return new Promise((resolve, reject) => {
+        pg.query(sql, params, (err, result) => err ? reject(err) : resolve(result.rows || []), readOnly);
+    });
+}
+
+async function getViewUpdatedAt (pg, view) {
+    const materializedViewDefinitionQuery = `select pg_get_viewdef(${view.reloid}, false) as definition`;
+
+    const readOnly = true;
+    const params = {};
+    const viewDefinition = await execQueryPromise({ pg, sql: materializedViewDefinitionQuery, params, readOnly });
+
+    const query = viewDefinition[0].definition;
+    const getAffectedTablesPromise = promisify(getAffectedTables);
+    const affectedTableView = await getAffectedTablesPromise(pg, query);
+
+    const getTablesMetadataPromise = promisify(getTablesMetadata);
+
+    const result = await getTablesMetadataPromise(pg, affectedTableView);
+
+    let latestTableMetadata = {};
+
+    for (const tableMetadata of result) {
+        if (!latestTableMetadata.updated_at) {
+            latestTableMetadata = tableMetadata;
+            continue;
+        }
+
+        const latestDate = new Date(latestTableMetadata.updated_at);
+        const tableMetadataDate = new Date(tableMetadata.updated_at);
+
+        if (tableMetadataDate.getTime() > latestDate.getTime()) {
+            latestTableMetadata = tableMetadata;
+        }
+    }
+
+    return latestTableMetadata.updated_at;
+}
+
+/* For foreign tables and materialized views to foreign tables
+   we need to do extra queries to extract the updated_at properly */
+async function setUpdateAtToNonRegularTables (pg, rows) {
     const getForeignTableUpdatedAtPromise = promisify(getForeignTableUpdatedAt);
+
     for (const row of rows) {
-        if (row.relkind === 'f') {
+        switch (row.relkind) {
+        case 'f': // foreign table
             row.updated_at = await getForeignTableUpdatedAtPromise(pg, row);
+            break;
+        case 'm': // materialized view
```

Comment:
I don't agree with this design decision, materialized view should be treated as regular tables to manage cache. Changes in the source table don't matter at all as long as you refresh the materialized view, so invalidating the cache won't mean anything. IMO, materialized views should have an entry in the local table updated when they are refreshed.

Reply:
It needs some analysis to understand what is the best approach. Going to the definition to know when it was updated doesn't make sense if the materialized view was not refreshed. We need to implement a way to save when a materialized view was updated.

Reply:
I'm going to revert this part as I'd like to move this forward and reevaluate it when we have a clear idea of what we need with Federated tables. Do you agree?

Reply:
Absolutely.